mri/utils/process_SWI

```diff
@@ -0,0 +1,127 @@
+        size of the resulting hanning filter
+    Returns
+    -------
+    imRef_homodyne: np.ndarray
+        the hanning filter matrix
+    """
+
+    nx, ny, nz = volume_shape
+
+    # create hanning filter 3D
+    hanning_1d = np.zeros([np.minimum(nx,ny),1])
+    hanning_1d[int(np.round(3 * nx / 10)):int(np.round(7 * nx / 10))] = \
+        np.expand_dims(np.hanning(int(np.round(7 * nx / 10))-int(np.round(3 * nx / 10))), 1)
+    hanning_2d = np.fft.fftshift(np.dot(hanning_1d, hanning_1d.T))
+    hanning_Nd = np.tile(np.expand_dims(hanning_2d, -1), (1,1,nz))
+
+    return hanning_Nd
+
+def get_SWI(imRef,  verbose=100) :
+    """
+        Parameters
+        ----------
+        image_ch: np.ndarray
+            Images reconstructed per channels in shape [Nch, Nx, Ny, Nz] in 3D or [Nch, Nx, Ny] in 2D
+        Returns
+        -------
+        SWI: np.ndarray
+            SWI image
+    """
+    if len(imRef.shape) == 3 :
+        image=imRef[:,:,:,np.newaxis]
+
+    # 1. get unique complex image
+    imTot = virtual_coil_reconstruction(imRef)
+
+    # 2. apply hanning filter on kspace and get get low-pass filtered phase image
+
+    hanning_filt = get_hanning_filter(imTot.shape)
+    imTot_filtered = np.fft.ifftn(np.fft.fftn(imTot)* hanning_filt)
+
+    # compute homodyne image
+    imTot_homodyne = imTot* np.exp(-1j * np.angle(imTot_filtered))
+
+    # 3. get phase mask
+    imTot_filt_phase = np.angle(imTot_homodyne)
+    pha_mask_raw = np.ones(imTot_filt_phase.shape)
+    pha_mask_raw[imTot_filt_phase > 0] = (
+            1 - imTot_filt_phase[imTot_filt_phase > 0] / np.pi)  # Phase mask type equation (4.21) in SWI Haacke
+    pha_mask_raw_final = pha_mask_raw ** 5  # Phase mask is applied between 4 and 5 times
+
+    # 4. compute SWI
+    swi_from_raw = np.abs(imTot) * pha_mask_raw_final
+
+    # display process
+    if verbose >= 100:
+        fig, ax = plt.subplots(nrows=2, ncols=3)
+        show_slice = int(imRef.shape[3]/2)
+
+        ax[0, 0].axis('off')
+        ax[0, 0].imshow(np.abs(imTot[:, :, show_slice].T), cmap='gray', aspect='equal')
+        ax[0, 0].title.set_text('Magnitude Image')
+        ax[1, 0].axis('off')
+        ax[1, 0].imshow(np.angle(imTot[:, :, show_slice].T), cmap='gray', aspect='equal')
+        ax[1, 0].title.set_text('Raw Phase Image ')
+        ax[1, 1].axis('off')
+        ax[1, 1].imshow(imTot_filt_phase[:, :, show_slice].T, cmap='gray', aspect='equal')
+        ax[1, 1].title.set_text('Filtered Phase Image')
+        ax[0, 1].axis('off')
+        ax[0, 1].imshow(np.abs(pha_mask_raw_final[:, :, show_slice].T), cmap='gray', aspect='equal')
+        ax[0, 1].title.set_text('Phase Mask')
+        ax[0, 2].axis('off')
+        ax[0, 2].imshow(np.abs(swi_from_raw[:, :, show_slice].T), cmap='gray', aspect='equal')
+        ax[0, 2].title.set_text('SW Image')
+        ax[1, 2].axis('off')
+        ax[1, 2].title.set_text('SW Image (mIP)')
+        plt.show()
+
+    return swi_from_raw
```

Comment:
Are we sure we dont want any other images? Like phase / phase mask etc?

Reply:
Well, we want to be able to display the process as I did in the function. Should it be appart from the function, on a separate script?